discovery-provider/tests/test_index_operations.py

```diff
@@ -101,39 +102,196 @@ def test_index_operations(app, client, celery_app, contracts):
+        "new_user_id": user_id_from_event
+    }
+
+@pytest.fixture(autouse=True)
+def cleanup():
+    yield
     remove_test_file(track_metadata_json_file)
+
+
+def test_index_operations(celery_app, celery_app_contracts):
+    """
+    Confirm indexing of user operations results in expected state change
+    """
+    task = celery_app.celery.tasks["update_discovery_provider"]
+    db = task.db
+    web3 = celery_app_contracts["web3"]
+
+    seed_data = seed_contract_data(task, celery_app_contracts, web3)
+    new_user_handle = seed_data["new_user_handle"]
+    new_user_id = seed_data["new_user_id"]
+
+    with db.scoped_session() as session:
+        # Catch up the indexer
+        current_block_query = session.query(Block).filter_by(is_current=True).all()
+        current_block = current_block_query[0].number if len(current_block_query) > 0 else 0
+        latest_block = web3.eth.getBlock("latest", True)
+        while current_block < latest_block.number:
+            # Process a bunch of blocks to make sure we covered everything
+            task.run()
+            current_block_query = session.query(Block).filter_by(is_current=True).all()
+            current_block = current_block_query[0].number if len(current_block_query) > 0 else 0
+
+        # Make sure the data we added is there
+        users = (
+            session.query(User)
+            .filter(User.handle == new_user_handle)
+            .all()
+        )
+        tracks = (
+            session.query(Track)
+            .filter(Track.owner_id == new_user_id)
+            .all()
+        )
+
+        assert len(users) > 0
+        assert len(tracks) > 0
+
+
+def test_index_operations_indexing_error(celery_app, celery_app_contracts, monkeypatch):
+    """
+    Confirm indexer throws IndexingError when the parser throws an error
+    """
+    task = celery_app.celery.tasks["update_discovery_provider"]
+    db = task.db
+    web3 = celery_app_contracts["web3"]
+
+    # Monkeypatch parse track event to raise an exception
+    def parse_track_event(*_):
+        raise Exception("Broken parser")
+    monkeypatch.setattr(src.tasks.tracks, "parse_track_event", parse_track_event)
+
+    seed_contract_data(task, celery_app_contracts, web3)
+
+    try:
+        with db.scoped_session() as session:
+            # Catch up the indexer
+            current_block_query = session.query(Block).filter_by(is_current=True).all()
+            current_block = current_block_query[0].number if len(current_block_query) > 0 else 0
+            latest_block = web3.eth.getBlock("latest", True)
+            while current_block < latest_block.number:
+                # Process a bunch of blocks to make sure we covered everything
+                task.run()
+                current_block_query = session.query(Block).filter_by(is_current=True).all()
+                current_block = current_block_query[0].number if len(current_block_query) > 0 else 0
+        assert False
+    except IndexingError:
```

Comment:
cc @jowlee  lmk if this is what you envisioned

Reply:
yeah this looks good, thanks for doing this!